### PR TITLE
Signature - Some organization, certificate revogation and tests

### PR DESCRIPTION
**Pull request by otaviobp**
- Reorganize code to take swupd specific code from signature.c
 - Add unit tests for signature verification
 - Check if OCSP is set as critical and reject the certificate if so, because OCSP isn't supported on swupd yet.

### CODE REVIEW

File: src/signature.c [hunk 1]
```diff
@@ -301,6 +301,76 @@ static X509 *get_cert_from_path(const char *certificate_path)
...
+{
+	AUTHORITY_INFO_ACCESS *info;
+	int n, i;
+
+	// Check if Authority Information Access is critical
+	if (!is_x509_ext_critical(cert, NID_info_access)) {
+		return 0;
+	}
+
+	debug("Authority Information Access is critical. Checking certificate revokation method\n");
```
Comment (castulo) [inline src/signature.c]: typo, it should be revocation not revokation

File: src/signature.c [hunk 2]
```diff
@@ -301,6 +301,76 @@ static X509 *get_cert_from_path(const char *certificate_path)
...
+		ACCESS_DESCRIPTION *ad = sk_ACCESS_DESCRIPTION_value(info, i);
+		if (OBJ_obj2nid(ad->method) == NID_ad_OCSP) {
+			error("OCSP uri found, but method not supported.\n");
+			return -1;
+		}
+	}
+	AUTHORITY_INFO_ACCESS_free(info);
+
+error:
+	error("Supported Authority Information Access methods not found in the certificate.");
```
Comment (castulo) [inline src/signature.c]: Missing \n?

File: src/signature.c [hunk 3]
```diff
@@ -301,6 +301,76 @@ static X509 *get_cert_from_path(const char *certificate_path)
...
+	info = X509_get_ext_d2i(cert, NID_info_access, NULL, NULL);
+	if (!info) {
+		goto error;
+	}
+
+	n = sk_ACCESS_DESCRIPTION_num(info);
+	for (i = 0; i < n; i++) {
+		ACCESS_DESCRIPTION *ad = sk_ACCESS_DESCRIPTION_value(info, i);
+		if (OBJ_obj2nid(ad->method) == NID_ad_OCSP) {
+			error("OCSP uri found, but method not supported.\n");
```
Comment (castulo) [inline src/signature.c]: This is kind of picky but I see many inconsistencies using "." at the end of messages.

File: test/unit/test_signature.c [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+	free(public_key);
+
+	return ret;
+}
+
+int main()
+{
+
+	if (set_up() < 0) {
+		print_error("Test set_up() error.\n");
```
Comment (castulo) [inline test/unit/test_signature.c]: print_error already include a \n, did you mean to add two \n to this message?

File: src/signature.c (hunk 1 above)
Comment (otaviobp) [reply src/signature.c]: I wrote revoke so many times yesterday that I got confused with that word. hahaha

**Comment by otaviobp** [comment]
fixing all comments

File: src/signature.c (hunk 1 above)
Comment (castulo) [reply src/signature.c]: it happens 😁

**Pull request merged**